### ISSUE TITLE
Question, multi task learning sequentially

### ISSUE THREAD

**Issue by lockder** [OpenNMT/OpenNMT-tf#299]
First, happy new year 💃 
I found some papers talking about having the same input embbeding but doing different trainings with different target objectives sequentally to share the inputs and improve the predicction.

for example:
- we create a part of speech tagging model
- then we want to predict the chunk of the sentences to split those sentences but we use the pos tag we predicted before as feature
- then we decide to create a dependency parser model, and we use the pos tag + chunked inputs as feature to help the dependency parser model
- then the ner we do the same thing, we use the predicted inputs from the other models as feature for the new model.

What is happening here is for the next model to be created we may use other pieces from others models from before but right now each new model, its independent from each other.
So the inputs can't be shared and doesn't get trained from each feature

Currenty we can't do that because we have only 1 target to predict and 1 optimizer.
**I'm not sure if could be posible to create something like the described before, were you share the word inputs for all the trainings but also each predicted model can be used as feature for the next model.**

So I'm not sure if could be posible to do something like this easily

**Comment by guillaumekln** [comment]
Hi,

I assume we expect the gradients to flow between tasks also? This would surely require some custom code. You could reuse some OpenNMT-tf functions as building blocks and build your own training code. It should not be that complicated if each task is properly encapsulated.

I will most likely never support something like this out-of-the-box as it makes everything more complicated.

**Comment by lockder** [comment]
yes, gradients from each task should affect the shared input words to train  them.
Thank you for your help I will try then
